### ISSUE TITLE
CageConfigurator can no longer be started

### ISSUE THREAD

**Issue by SailReal** [SharkCagey/HTWG_shark_cage#104]
Issue with the `CageConfigurator` on the secure desktop introduced by @bencikpeter:
> It switches mi to the desktop, labeler loads, but no window appears and when I close the cage, labelel finishes but I stay on a desktop and am forced to restart the machine.
   other apps work just fine.

More information: https://github.com/SharkCagey/HTWG_shark_cage/pull/99#issuecomment-411865332

**Comment by SailReal** [comment]
@bencikpeter is it possible that we have a problem with any kind of permissions running the `CageConfigurator` on the different desktop?

**Comment by bencikpeter** [comment]
Maybe we can try to run something else that needs admin privileges in cage? just to see what happens

**Comment by DonatJR** [comment]
@bencikpeter Just did that and I am indeed having the same issues with both `HWMonitor` and `Nvidia Inspector` - both asking for elevation when started normally, bot not showing up on the secure desktop.

**Comment by bencikpeter** [comment]
@DonatJR And should we fix that? 

As @SailReal  pointed out, it worked before [db7a14e549c8b64ee6a1d7fb8c01448458e7548b](https://github.com/SharkCagey/HTWG_shark_cage/commit/db7a14e549c8b64ee6a1d7fb8c01448458e7548b) The reason that it worked though was, that before that moment, all caged aplications were started as `NT AUTHORITY` ... therefore no UAC prompt, since they were by default elevated. From that poit on, caged applications are started in security context of logged on user... I would say that the bug with displaying UAC has always existed and has nothing to do with changes in that particular commit, they just surfaced it.

**I see two solutions:** 

1. Figure out a way how to display UAC prompt on secure desktop (which might be a little overkill, since UAC I belive is started on a separate (`WINLOGON `) desktop on it´s own...)

2. Just state in a documentation that apps invoking UAC prompt are not compatible with Shark Cage and  figure out how to avoid starting `CageChooser` elevated

**Comment by DonatJR** [comment]
I think we should most definitely fix it, yes. Not being compatible with UAC prompts is kind of a big compromise, imo. Also, the user can't reliably predict which application causes an UAC prompt and if it happens the user is stuck with rebooting / logging out.

A third solution: Have a boolean flag in the config if the process should be started elevated like before (also requires previous knowledge of the application from the user).

I'm currently hoping the problem will resolve itself after #102 as the UAC prompt is started with system credentials and the system was previously not allowed to do anything on our desktop.

**Comment by DonatJR** [comment]
#102 should fix this for now, but we can keep this issue open to improve it in the future

**Comment by SailReal** [comment]
In my opinion this can be closed because it is fixed. 
If we can improve something (do we) we can create an new issue for that?

**Comment by DonatJR** [comment]
alright, let's close it with #102. I created a new issue for improving the elevation dialog prompt: #110

**Comment by SailReal** [comment]
Fixed with: https://github.com/SharkCagey/HTWG_shark_cage/pull/105
